tests/test_serialization.py

```diff
@@ -0,0 +1,67 @@
+from .sample_requests import sample_requests, BASE_URL
+from .helpers import clean_headers, clean_body
+
+import pytest
+from requests_flask_adapter import Session
+from tempfile import NamedTemporaryFile
+
+import catchall_server
+from serializable_request import Request
+
+out_req = None
+
+
+@pytest.fixture
+def session():
+    Session.register(BASE_URL, catchall_server.app)
+    return Session()
+
+
+def pseudo_forward():
+    '''
+    Send request to flask.
+    Recreate request using flask internal request context.
+
+    This tests if we use flask.request correctly.
+    '''
+    global out_req
+    out_req = Request.from_flask_request().as_requests_request()
+    return {}, 200
+
+
+def pseudo_forward_with_file():
+    '''
+    Send request.
+    Recreate request from flask internal request context.
+    Save request to file.
+    Read request from this file.
+
+    Assuming pseudo_forward finds no errors, this tests if we correctly
+    serialize/deserialize the request.
+    '''
+    global out_req
```

Comment:
Yes, i think this will mess up the parallel tests, good point.

This is a function used internally by `catchall_server` and it is supposed to return something that Flask could turn into a response. I'd prefer not to mess with this interface for the sake of tests, but I guess I could just add a wrapper function.

Reply:
832f1e9